packages/runtime/test/transport/identityDeletionProcess.test.ts

```diff
@@ -43,16 +50,37 @@ afterEach(async () => {
 });
 
 describe("IdentityDeletionProcess", () => {
+    test("should start and cancel an IdentityDeletionProcess using the Backbone Admin Api", async function () {
+        const initiatedIdentityDeletionProcess = await startIdentityDeletionProcessFromBackboneAdminApi(transportService, accountAddress);
+        expect(initiatedIdentityDeletionProcess.status).toStrictEqual(IdentityDeletionProcessStatus.WaitingForApproval);
+
+        await expect(eventBus).toHavePublished(IdentityDeletionProcessStatusChangedEvent, (e) => e.data.id === initiatedIdentityDeletionProcess.id);
+        await expect(eventBus).toHavePublished(IdentityDeletionProcessStatusChangedEvent, (e) => e.data.status === IdentityDeletionProcessStatus.WaitingForApproval);
+
+        await transportService.identityDeletionProcesses.approveIdentityDeletionProcess();
+
+        const result = await cancelIdentityDeletionProcessFromBackboneAdminApi(transportService, accountAddress, initiatedIdentityDeletionProcess.id.toString());
+        expect(result).toBeSuccessful();
+        const identityDeletionProcess = result.value;
+        expect(identityDeletionProcess.status).toStrictEqual(IdentityDeletionProcessStatus.Cancelled);
+
+        await expect(eventBus).toHavePublished(IdentityDeletionProcessStatusChangedEvent, (e) => e.data.id === initiatedIdentityDeletionProcess.id);
```

Comment:
You either need to reset the event bus in between create/approve and cancel or move the check for the status in the same callback, as the toHavPublished will check all the event since the last reset.

And as the event name and the IDP id are the same this check will always "succeed" with the event from the creation